### PR TITLE
[v2][ios11, ios12] Add searchBar option for topBar

### PR DESCRIPTION
**Pull request by ptmt**
Fully native SearchBar and UISearchController, available in iOS 11
This is a captured Detox test demonstrating event handling:
jun-05-2018 11-56-40

Scrolling:
jun-05-2018 12-06-30

I don't think it's time to write the docs yet, wanting to battle-test it more and polish the API.

### CODE REVIEW

File: lib/src/components/ComponentWrapper.tsx [hunk 1]
```diff
@@ -1,16 +1,24 @@
-import * as React from 'react';
-import * as  _ from 'lodash';
-import * as ReactLifecyclesCompat from 'react-lifecycles-compat';
+import * as React from "react";
```
Comment (guyca) [inline lib/src/components/ComponentWrapper.tsx]: Can you please auto formatting and keep only actual changes in the PR? Thanks 👍

**Comment by guyca** [comment]
@ptmt Thanks for the PR, looks great 👍 
Can you please update the docs as well?

File: lib/src/components/ComponentWrapper.tsx (hunk 1 above)
Comment (ptmt) [reply lib/src/components/ComponentWrapper.tsx]: oh, sorry. Sure

Comment (guyca) [reply lib/src/components/ComponentWrapper.tsx]: No worries, thanks :)

**Comment by ptmt** [comment]
Added a few lines to docs. Not sure why Android is failing.

**Comment by birkir** [comment]
Nice one, but are you running prettier? I always find it hard to read PR's where there is code changes irrelevant to the PR, just to make it pretty.

File: docs/docs/events.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Called each time this component appears on screen (attached to the view heirarch
 ```js
 class MyComponent extends Component {
   componentDidAppear() {
-    
+
```
Comment (ptmt) [inline docs/docs/events.md]: this is not even the Prettier, turns out that something else in Atom applies these changes 😓

Comment (birkir) [reply docs/docs/events.md]: Damn, that's probably editorconfig

Comment (ptmt) [reply docs/docs/events.md]: Pardon me, I haven't touched Atom and JS for a few months. I revert most of changes, except these empty lines + I updated tests

**Comment by ptmt** [comment]
Thanks for your time! Just to be clear, is there anything needed from my side?

**Comment by yogevbd** [comment]
Looks awesome! thanks

**Comment by ptmt** [comment]
oh, okay. 
looking into it (it seems that some combination of properties is needed currently in order to make it work properly)

**Comment by ptmt** [comment]
@birkir could you elaborate and/or create a separate issue? Let's polish this feature together :)

**Pull request merged**